packages/wonder-blocks-data/components/intercept-cache.md

```diff
@@ -0,0 +1,101 @@
+(options: TOptions, cacheEntry: ?$ReadOnly<CacheEntry<TData>>) => ?$ReadOnly<CacheEntry<TData>>
+```
+
+`options` are the options that would configure a data request that leads to the
+cached result.
+`cacheEntry` is the existing entry in the data cache.
+
+If the method returns `null`, the default behavior occurs. This means that if
+`cacheEntry` has a value, that will be used; otherwise, a request will be
+made for data via the relevant handler assigned to the `Data` component being
+intercepted.
+
+```jsx
+import {Body, BodyMonospace} from "@khanacademy/wonder-blocks-typography";
+import {View} from "@khanacademy/wonder-blocks-core";
+import {InterceptCache, Data, RequestHandler} from "@khanacademy/wonder-blocks-data";
+import {Strut} from "@khanacademy/wonder-blocks-layout";
+import Color from "@khanacademy/wonder-blocks-color";
+import Spacing from "@khanacademy/wonder-blocks-spacing";
+
+class MyHandler extends RequestHandler {
+    constructor() {
+        super("INTERCEPT_CACHE_HANDLER");
+    }
+
+    /**
+     * fulfillRequest should not get called as we already have data cached.
+     */
+    fulfillRequest(options) {
+        throw new Error(
+            "If you're seeing this error, the examples are broken.",
+        );
+    }
+
+    shouldRefreshCache(options, cachedEntry) {
+        /**
+         * For our purposes, the cache never needs a refresh.
+         */
+        return false;
+    }
+}
+
+const handler = new MyHandler();
+const getEntryIntercept = function(options) {
+    if (options === "ERROR") {
+        return {error: "I'm an ERROR from the cache interceptor"};
+    }
+
+    return {data: "I'm DATA from the cache interceptor"};
+};
+
+<InterceptCache handler={handler} getEntry={getEntryIntercept}>
+    <View>
+        <Body>This intercepted cache has data!</Body>
+        <Data handler={handler} options={"DATA"}>
+            {({loading, data}) => {
+                if (loading) {
+                    return "If you see this, the example is broken!";
+                }
+
+                return (
+                    <BodyMonospace>{data}</BodyMonospace>
+                );
+            }}
+        </Data>
+    </View>
+    <Strut size={Spacing.small} />
+    <View>
+        <Body>This intercepted cache has error!</Body>
+        <Data handler={handler} options={"ERROR"}>
+            {({loading, error}) => {
+                if (loading) {
+                    return "If you see this, the example is broken!";
+                }
+
+                return (
+                    <BodyMonospace style={{color: Color.red}}>ERROR: {error}</BodyMonospace>
+                );
+            }}
+        </Data>
+    </View>
+</InterceptCache>
+```
+
+This component can also be useful if you want to avoid a data request because
+a parent component already has the data you need. Using this component, you
+can map already cached data of one request (or a portion thereof) to another
+without affecting the cache itself.
```

Comment:
There simpler ways of passing data from a component down to a descendant.  When would it make sense to use this method over other methods?

Reply:
Not sure I understand again. This is preferable when a component is using `Data` and you want to override how it gets its data.

Otherwise, you'd have to edit the component to take a prop just for that purpose and then have multiple rendering routes based on the presence of that prop. These components were created to avoid needing to do that kind of change to components when using the data framework. Of course, it would still be valid. However, as soon as that component is x components deep, you've got to put that prop on x components.

Reply:
It sounds like an alternative to using a React Context.  Are there benefits of using this over React Context?

Reply:
This is using React Context. The entire point here is to avoid having to write a custom react context for every single component that uses `Data`.